src/ui/hook/usePrevious.ts

```diff
@@ -0,0 +1,9 @@
+import { useRef, useEffect } from 'react'
+
+export const usePrevious = <T>(value: T): T | undefined => {
```

Comment:
I'm wondering if it's really useful to make it as a hook, I think you could keep your ref in the stepper progress file.

Reply:
This is a widely used hook pattern that might be reusable somewhere else in the app. Until we need it somewhere else, keeping it in the stepper progress file is fine too.

Reply:
You're right it makes sense 👍